src/analysis/normed_space/pi_Lp.lean

```diff
@@ -144,64 +260,101 @@ structure and the bornology by the product ones using this pseudometric space,
+    begin
+      apply ennreal.rpow_le_rpow _ (one_div_nonneg.2 $ (zero_le_one.trans h)),
+      exact finset.single_le_sum (λ i hi, (bot_le : (0 : ℝ≥0∞) ≤ _)) (finset.mem_univ i)
+    end }
 end
 
-lemma antilipschitz_with_equiv_aux :
-  antilipschitz_with ((fintype.card ι : ℝ≥0) ^ (1/p)) (pi_Lp.equiv p β) :=
+lemma antilipschitz_with_equiv_aux [fact (1 ≤ p)] :
+  antilipschitz_with ((fintype.card ι : ℝ≥0) ^ (1 / p).to_real) (pi_Lp.equiv p β) :=
 begin
-  have pos : 0 < p := lt_of_lt_of_le zero_lt_one fact_one_le_p.out,
-  have nonneg : 0 ≤ 1 / p := one_div_nonneg.2 (le_of_lt pos),
-  have cancel : p * (1/p) = 1 := mul_div_cancel' 1 (ne_of_gt pos),
-  assume x y,
-  simp [edist, -one_div],
-  calc (∑ i, edist (x i) (y i) ^ p) ^ (1 / p) ≤
-  (∑ i, edist (pi_Lp.equiv p β x) (pi_Lp.equiv p β y) ^ p) ^ (1 / p) :
-  begin
-    apply ennreal.rpow_le_rpow _ nonneg,
-    apply finset.sum_le_sum (λ i hi, _),
-    apply ennreal.rpow_le_rpow _ (le_of_lt pos),
-    exact finset.le_sup (finset.mem_univ i)
-  end
-  ... = (((fintype.card ι : ℝ≥0)) ^ (1/p) : ℝ≥0) *
-    edist (pi_Lp.equiv p β x) (pi_Lp.equiv p β y) :
-  begin
-    simp only [nsmul_eq_mul, finset.card_univ, ennreal.rpow_one, finset.sum_const,
-      ennreal.mul_rpow_of_nonneg _ _ nonneg, ←ennreal.rpow_mul, cancel],
-    have : (fintype.card ι : ℝ≥0∞) = (fintype.card ι : ℝ≥0) :=
-      (ennreal.coe_nat (fintype.card ι)).symm,
-    rw [this, ennreal.coe_rpow_of_nonneg _ nonneg]
-  end
+  intros f g,
+  unfreezingI { rcases p.dichotomy with (rfl | h) },
+  { simp only [edist_eq_supr, ennreal.div_top, ennreal.zero_to_real, nnreal.rpow_zero,
+      ennreal.coe_one, equiv_apply', one_mul, supr_le_iff],
+    exact λ i, finset.le_sup (finset.mem_univ i), },
+  { have pos : 0 < p.to_real := zero_lt_one.trans_le h,
+    have nonneg : 0 ≤ 1 / p.to_real := one_div_nonneg.2 (le_of_lt pos),
+    have cancel : p.to_real * (1/p.to_real) = 1 := mul_div_cancel' 1 (ne_of_gt pos),
+    rw [edist_eq_sum pos, ennreal.to_real_div 1 p],
+    simp only [edist, equiv_apply', ←one_div, ennreal.one_to_real],
+    calc (∑ i, edist (f i) (g i) ^ p.to_real) ^ (1 / p.to_real) ≤
+    (∑ i, edist (pi_Lp.equiv p β f) (pi_Lp.equiv p β g) ^ p.to_real) ^ (1 / p.to_real) :
+    begin
+      apply ennreal.rpow_le_rpow _ nonneg,
+      apply finset.sum_le_sum (λ i hi, _),
+      apply ennreal.rpow_le_rpow _ (le_of_lt pos),
+      exact finset.le_sup (finset.mem_univ i)
+    end
+    ... = (((fintype.card ι : ℝ≥0)) ^ (1 / p.to_real) : ℝ≥0) *
+      edist (pi_Lp.equiv p β f) (pi_Lp.equiv p β g) :
+    begin
+      simp only [nsmul_eq_mul, finset.card_univ, ennreal.rpow_one, finset.sum_const,
+        ennreal.mul_rpow_of_nonneg _ _ nonneg, ←ennreal.rpow_mul, cancel],
+      have : (fintype.card ι : ℝ≥0∞) = (fintype.card ι : ℝ≥0) :=
+        (ennreal.coe_nat (fintype.card ι)).symm,
+      rw [this, ennreal.coe_rpow_of_nonneg _ nonneg]
+    end }
 end
 
-lemma aux_uniformity_eq :
+lemma aux_uniformity_eq [fact (1 ≤ p)] :
```

Comment:
Why did you move `[fact (1 ≤ p)]` from the `variables` to lemma statements?

Reply:
Because many of the constructions we want to define also for `p < 1`. In the previous iteration, `[fact (1 ≤ p)]` was a named variable with several `include` and `omit`s throughout the file. I find that kind of source much harder reading, so I just chose to include it per declaration until after the norm is defined (when I then make it a variable for the rest of the file).

Reply:
To me that says we should put the bits that work for all `p` before what's restricted to `1 ≤ p`. At least that means that your change is a style one, so might be better reverting it to reduce the diff.

Reply:
Moving the definitions of `edist`, `dist` and `norm` all to the beginning and then later doing the stuff for `[fact (1 ≤ p)]` I think makes the most sense, thanks.